nvflare/app_common/widgets/log_streaming.py

```diff
@@ -0,0 +1,117 @@
+                stream_ctx={LogConst.CLIENT_NAME: client_name, LogConst.JOB_ID: job_id},
+                targets=["server"],
+                file_name=error_log_path,
+                fl_ctx=fl_ctx,
+            )
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == self.event_type:
+            if self.should_report_error_log:
+                t = threading.Thread(target=self._stream_error_log_file, args=(fl_ctx,), daemon=True)
+                t.start()
+                client_name = fl_ctx.get_prop(FLContextKey.CLIENT_NAME)
+                job_id = fl_ctx.get_prop(FLContextKey.CURRENT_JOB_ID)
+                self.log_info(fl_ctx, f"Started streaming error log file for {client_name} for job {job_id}")
+
+
+class LogReceiver(Widget):
+    def __init__(self, log_channels=None):
+        """Receives log data.
+
+        By default, it receives error logs with the channel error_logs to save logs of type ERRORLOG. If adding
+        additional log types, make sure nvflare.apis.storage.ComponentPrefixes has the corresponding log type.
+
+        Args:
+            log_channels: dict of channel to log type mapping
+        """
+        super().__init__()
+        if log_channels is None:
+            log_channels = {"error_logs": "ERRORLOG"}
+        self.log_channels = log_channels
+
+    def process_log(self, stream_ctx: StreamContext, fl_ctx: FLContext):
+        """Process the streamed log file."""
+        peer_ctx = fl_ctx.get_peer_context()
+        assert isinstance(peer_ctx, FLContext)
+        peer_name = peer_ctx.get_identity_name()
+        channel = FileStreamer.get_channel(stream_ctx)
+        topic = FileStreamer.get_topic(stream_ctx)
+        rc = FileStreamer.get_rc(stream_ctx)
+        if rc != ReturnCode.OK:
+            self.log_error(
+                fl_ctx,
+                f"Error in streaming log file from {peer_name} on channel {channel} and topic {topic} with rc {rc}",
+            )
+            return
+        file_location = FileStreamer.get_file_location(stream_ctx)
+        self.log_info(fl_ctx, f"File location: {file_location}")
+        with open(file_location, "r") as f:
+            log_contents = f.read()
+        client = stream_ctx.get(LogConst.CLIENT_NAME)
+        job_id = stream_ctx.get(LogConst.JOB_ID)
+        job_manager = fl_ctx.get_engine().get_component(SystemComponents.JOB_MANAGER)
+        log_type = self.log_channels.get(channel, "UNKNOWN")
+        self.log_info(fl_ctx, f"Saving {log_type} from {client} for {job_id}")
+        job_manager.set_log(job_id, log_contents, client, log_type, fl_ctx)
+
+    def handle_event(self, event_type: str, fl_ctx: FLContext):
+        if event_type == EventType.SYSTEM_START:
+            for channel in self.log_channels.items():
+                FileStreamer.register_stream_processing(
+                    fl_ctx, channel=channel, topic=LOG_STREAM_EVENT_TYPE, stream_done_cb=self.process_log
+                )
```

Comment:
We could also say this LogSender and LogReceiver is always using the same channel but for different types of logs they send/receive they use different "topic"?

Now is the same "topic" but different "channel" I guess we just need to explain what we prefer in the docstring

Reply:
You should use "*" to be the topic such that all logs will be handled by the same CB.
Let's use the same predefined channel for all log streaming. Topic is not important here, since any log specific info will be in the stream_ctx.